protocol/encryption/persistence.go

```diff
@@ -729,34 +730,34 @@ func (s *sqliteSessionStorage) Load(id []byte) (*dr.State, error) {
 }
 
 type HRCache struct {
-	GroupID []byte
-	KeyID   uint32
-	Key     []byte
-	Hash    []byte
-	SeqNo   uint32
+	GroupID         []byte
+	KeyID           []byte
+	DeprecatedKeyID uint32
+	Key             []byte
+	Hash            []byte
+	SeqNo           uint32
 }
 
 // GetHashRatchetKeyByID retrieves a hash ratchet key by group ID and seqNo.
 // If cache data with given seqNo (e.g. 0) is not found,
 // then the query will return the cache data with the latest seqNo
-func (s *sqlitePersistence) GetHashRatchetKeyByID(groupID []byte, keyID uint32, seqNo uint32) (*HRCache, error) {
-	stmt, err := s.DB.Prepare(
-		`WITH input AS (
-       select ? AS group_id, ? AS key_id, ? as seq_no
+func (s *sqlitePersistence) GetHashRatchetKeyByID(ratchet *HashRatchetKeyCompatibility, seqNo uint32) (*HRCache, error) {
+	stmt, err := s.DB.Prepare(`WITH input AS (
+       select ? AS group_id, ? AS key_id, ? as seq_no, ? AS old_key_id
      ),
      cec AS (
        SELECT e.key, c.seq_no, c.hash FROM hash_ratchet_encryption e, input i
-			 LEFT JOIN hash_ratchet_encryption_cache c ON e.group_id=c.group_id AND e.key_id=c.key_id
-       WHERE e.key_id=i.key_id AND e.group_id=i.group_id),
+			 LEFT JOIN hash_ratchet_encryption_cache c ON e.group_id=c.group_id AND (e.hash_id=c.key_id OR e.key_id=c.key_id)
```

Comment:
Shouldn't it be? :thinking: 
```suggestion
			 LEFT JOIN hash_ratchet_encryption_cache c ON e.group_id=c.group_id AND (e.hash_id=c.hash OR e.key_id=c.key_id)
```

Reply:
hash is the current hash of original key + seqNo (i.e the original key hashed `sedNo` key), here we want to match with the old key_id format (deprecated_key_id as per your suggestion) and with the new format (`hash_id`), as we want to get the original key I believe

Reply:
I got confused with hash_id